### PR TITLE
added skip-nav to missing pages

### PR DESCRIPTION
**Pull request by Wu-Bowen**
## Description

Added reach-skip-nav IDs to main elements that were missing them.

## Related Issues

Related to issue #21565

### CODE REVIEW

**Comment by Wu-Bowen** [comment]
Hi @tesseralis,
Sorry about the component part, I must have missed the comment in the original issue. I changed all the SkipNavContent on the aforementioned pages. I had a question on where some of these pages are used. For example, the 404 page and the ecosystem page are easy enough to find, but for the template pages, I added the component, but I do not know where on the website I can go to test if they are working.

**Comment by tesseralis** [comment]
One thing I noticed from doing some research on skip-navs (ref: [WebAIM](https://webaim.org/techniques/skipnav/) [a11y-project](https://a11yproject.com/posts/skip-nav-links/)) was that they tend to be attached to the `<main>` tag. But Reach's `<SkipNavComponent>` only renders a div.

After talking with @madalynrose, I think the best solution is to roll out our own `<MainContent>` component that attaches the ID prop:

```jsx
function MainContent({ children, …props }) {
  <main id="reach-skip-nav" {…props}>
    {children}
  </main>
}
```

And then we can use this component in all our pages, replacing the raw reference to ID.

**Comment by tesseralis** [comment]
As for testing the template pages, check the different files under `src/utils/node`. These should tell you which pages and urls are associated with which templates. For example, in `utils/node/starters`:

```js
  starters.forEach((node, index) => {
    createPage({
      path: `/starters${node.fields.starterShowcase.slug}`,
      component: slash(starterTemplate),
      context: {
        slug: node.fields.starterShowcase.slug,
      },
    })
  })
```

 we see that `template-starter-page` is being used to create pages with the hash `/starters/{slug}`. So we know it's responsible for making pages like: https://www.gatsbyjs.org/starters/draftbox-co/gatsby-ghost-balsa-starter/

You can also use file search find where different packages are imported. For example, if you search for `template-starter-page` you should point you to the file in `utils/node` that uses that template.

**Comment by Wu-Bowen** [comment]
@tesseralis Just to understand, you want us to create a MainContent component that will be imported to all the pages. Then, to use this component instead of the id="reach-skip-nav"?

**Comment by tesseralis** [comment]
> @tesseralis Just to understand, you want us to create a MainContent component that will be imported to all the pages. Then, to use this component instead of the id="reach-skip-nav"? 

@Wu-Bowen correct!

**Comment by Wu-Bowen** [comment]
Hi @tesseralis, we added a new MainContent component and tested it on the pages that you had listed above. Please let us know if this was what you were looking to do and whether or not we should implement it for the rest of the pages.

File: www/src/pages/404.js [hunk 1]
```diff
@@ -2,18 +2,21 @@ import React from "react"
 import Container from "../components/container"
 import Link from "../components/localized-link"
 import FooterLinks from "../components/shared/footer-links"
+import Main from "../components/main-content"
 
 export default function FourOhFour() {
   return (
     <>
```
Comment (tesseralis) [inline www/src/pages/404.js:9]: We don't need this fragment any more since everything is wrapped in a `<Main>` tag.

**Comment by tesseralis** [comment]
@Wu-Bowen out of curiousity, what command did you run to format `www`? It's picking up some formatting errors that weren't picked up before and I'm worried that our linters might be broken.

**Comment by Wu-Bowen** [comment]
> @Wu-Bowen out of curiousity, what command did you run to format `www`? It's picking up some formatting errors that weren't picked up before and I'm worried that our linters might be broken.

Yea I noticed that, but I didn't know exactly which files were changed, so I couldn't revert it. I believe I ran yarn prettier --write src/**/*.js. Sorry if this was the wrong command, I wasn't too sure what Gatsby's built-in formatter was.

**Comment by tesseralis** [comment]
@Wu-Bowen Ah, I see. It appears that our linter hasn't been working for www for the past couple of months. I created a PR to fix that here: #24133.

**Comment by tesseralis** [comment]
So for now I think we can do one of two things:

1. wait for that PR to be merged.
2. undo your formatting changes and try to run prettier on *only* the files that you are changing.

If you'd rather get unblocked, feel free to do 2. But if you feel that's too much work already and would rather wait, that's fine as well.

**Comment by tesseralis** [comment]
@Wu-Bowen my eslint PR just got through! Could you fix the conflicts and update this branch? Thanks!

File: www/src/components/main-content.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from "react"
+
+const MainContent = ({ children, ...props }) => {
```
Comment (tesseralis) [inline www/src/components/main-content.js:3]: Prefer

```
export default function MainContent(…)
```

We've been inconsistent in the past, but going forward we should give components names: https://twitter.com/dan_abramov/status/1255229440860262400

File: www/src/pages/404.js [hunk 3]
```diff
@@ -14,6 +15,6 @@ export default function FourOhFour() {
         </Link>
       </Container>
       <FooterLinks />
-    </>
+    </Main>
```
Comment (tesseralis) [inline www/src/pages/404.js:18]: `<FooterLinks>` should not be nested in `<main>`: https://stackoverflow.com/questions/20470345/should-the-header-and-footer-tags-be-inside-the-main-tag

In general, the components should probably be right next to each other.

File: www/src/components/main-content.js (hunk 2 above)
Comment (tesseralis) [reply www/src/components/main-content.js:3]: In addition, could you write a comment that describes what this component is used for? e.g.:

```
/**
 * <main> content wrapper that should be included in every page on the site.
 */
```

**Comment by LekoArts** [comment]
Hi @Wu-Bowen 👋 Do you have time to make the requested changes and solve the merge conflict? :)

**Comment by freiksenet** [comment]
Hi! 

I'm going to close this now, as this appears to have been abandoned. Feel free to reopen this PR if you find time to address the comments.

Thanks!

**Pull request closed without merging**